cluster/images/etcd/attachlease.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+)
+
+func main() {
+	var etcdAddress string
+	var ttlKeysPrefix string
+	var leaseDuration time.Duration
+	flag.StringVar(&etcdAddress, "etcd-address", "", "Etcd address")
+	flag.StringVar(&ttlKeysPrefix, "ttl-keys-prefix", "", "Prefix for TTL keys")
+	flag.DurationVar(&leaseDuration, "lease-duration", time.Hour, "Lease duration")
```

Comment:
uh, you do integer division and cast to int64. 500ms will turn into 0.

On Wed, Aug 24, 2016 at 11:12 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In cluster/images/etcd/attachlease.go
> <https://github.com/kubernetes/kubernetes/pull/31242#discussion_r76109399>
> :
>
> > +	"flag"
> > +	"strings"
> > +	"time"
> > +
> > +	"github.com/coreos/etcd/clientv3"
> > +	"github.com/golang/glog"
> > +	"golang.org/x/net/context"
> > +)
> > +
> > +func main() {
> > +	var etcdAddress string
> > +	var ttlKeysPrefix string
> > +	var leaseDuration time.Duration
> > +	flag.StringVar(&etcdAddress, "etcd-address", "", "Etcd address")
> > +	flag.StringVar(&ttlKeysPrefix, "ttl-keys-prefix", "", "Prefix for TTL keys")
> > +	flag.DurationVar(&leaseDuration, "lease-duration", time.Hour, "Lease duration")
>
> That's not true - for flag duration you can pass e.g. 10m or 1h or 30s or
> even 500ms - it actually doesn't have seconds granularity.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31242/files/1ff93b535adac0175574f6085e02d434121a8955#r76109399>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglr86wID31MbZ-Pb4IgLzNoWiPUtwks5qjImBgaJpZM4Jq7E_>
> .
>